api/routes/project.py

```diff
@@ -56,61 +69,85 @@ async def create_project(
 @router.get('/seqr/all', operation_id='getSeqrProjects')
 async def get_seqr_projects(connection: Connection = get_projectless_db_connection):
     """Get SM projects that should sync to seqr"""
-    ptable = ProjectPermissionsTable(connection)
-    return await ptable.get_seqr_projects()
+    projects = connection.all_projects()
+    return [p for p in projects if p.meta and p.meta.get('is_seqr')]
 
 
 @router.post('/{project}/update', operation_id='updateProject')
 async def update_project(
-    project: str,
     project_update_model: dict,
-    connection: Connection = get_projectless_db_connection,
+    connection: Connection = get_project_db_connection(
+        {ProjectMemberRole.project_admin}
+    ),
 ):
     """Update a project by project name"""
     ptable = ProjectPermissionsTable(connection)
+
+    # Updating a project additionally requires the project creator permission
+    await ptable.check_project_creator_permissions(author=connection.author)
+
+    project = connection.project
+    assert project
     return await ptable.update_project(
-        project_name=project, update=project_update_model, author=connection.author
+        project_name=project.name, update=project_update_model, author=connection.author
     )
 
 
 @router.delete('/{project}', operation_id='deleteProjectData')
 async def delete_project_data(
-    project: str,
     delete_project: bool = False,
-    connection: Connection = get_projectless_db_connection,
+    connection: Connection = get_project_db_connection(
+        {ProjectMemberRole.project_admin}
+    ),
 ):
     """
     Delete all data in a project by project name.
     Can optionally delete the project itself.
     Requires READ access + project-creator permissions
     """
     ptable = ProjectPermissionsTable(connection)
-    p_obj = await ptable.get_and_check_access_to_project_for_name(
-        user=connection.author, project_name=project, readonly=False
+
+    assert connection.project
+
+    # Allow data manager role to delete test projects
+    data_manager_deleting_test = (
```

Comment:
Sorry, I'm not quite following how we would do this from cpg-infra? For context - this change is related to the discussion where Amy mentioned it would be handy for data managers to be able to clear out test projects https://centrepopgen.slack.com/archives/C020PDNH3N0/p1717389879026539

Reply:
Oh I think I might have just understood, we could give "data-managers" as defined in cpg-infrastructure some sort of project admin role for test projects. Rather than having a data manager role in metamist. That does sound like a better way of handling it. At the moment project admin in kind of a pseudo-role that is set by being in the `project-creators` group, but it would be possible to make it a role that can be set in the project_members table too.

Reply:
It's creeping outside the scope a bit, but I wonder if we should change it in the generated API to not provide an auth token if local / test? Then we wouldn't need to re-arrange here, and would probably speed up local testing by not always generating this token which does take some amount of time.

https://github.com/populationgenomics/metamist/blob/846c6b80006e0264c1153998bb825f0804bd68c9/openapi-templates/configuration.mustache#L502-L503